### PR TITLE
Add docstrings to `patroni.async_executor`

### PR DESCRIPTION
**Pull request by barthisrael**
References: PAT-120.

### CODE REVIEW

File: patroni/async_executor.py [hunk 1]
```diff
@@ -83,14 +128,32 @@ def schedule(self, action: str) -> Optional[str]:
...
     def reset_scheduled_action(self) -> None:
+        """Unschedule a previously scheduled action, if any.
+
+        .. note::
+            Must be called once the scheduled task finishes or is cancelled.
+        """
         with self._scheduled_action_lock:
             self._scheduled_action = None
 
     def run(self, func: Callable[..., Any], args: Tuple[Any, ...] = ()) -> Optional[bool]:
```
Comment (CyberDem0n) [inline patroni/async_executor.py:144]: I think that I did it wrong, the `func` could return anything, not necessarily `bool`, and the [docstring](https://github.com/zalando/patroni/pull/2704/files#diff-e79a66f0758d9f820b21814a9979d9067d4fa90b63ac1994376743a164f683aeR154-R155) confirms it.
```suggestion
    def run(self, func: Callable[..., Any], args: Tuple[Any, ...] = ()) -> Optional[Any]:
```

Comment (barthisrael) [reply patroni/async_executor.py]: Good catch!

File: patroni/async_executor.py [hunk 2]
```diff
@@ -13,52 +14,83 @@ class CriticalTask(object):
...
 class AsyncExecutor(object):
+    """Asynchronous executor of (long) tasks.
+
+    :ivar critical_task: a :class:`CriticalTask` instance to handle execution of critical background tasks.
+    """
 
     def __init__(self, cancellable: CancellableSubprocess, ha_wakeup: Callable[..., None]) -> None:
+        """Create a new instance of :class:`AsyncExecutor`.
+
+        Configure the given *cancellable* and *ha_wakefup*, initializes the control attributes, and instantiate the lock
```
Comment (hughcapet) [inline patroni/async_executor.py:88]:
```suggestion
        Configure the given *cancellable* and *ha_wakeup*, initializes the control attributes, and instantiate the lock
```

**Comment by hughcapet** [comment]
👍

**Comment by CyberDem0n** [comment]
:+1:

**Pull request merged**